src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py

```diff
@@ -21,13 +22,53 @@
     try_float_or_none,
     try_int,
     try_int_or_none,
+    try_str_from_series,
+    try_str_from_series_or_none,
 )
 
 BLOCKS_LINE_REGEX = r"^##BLOCKS=\s*(\d+)$"
 END_LINE_REGEX = "~End"
 EXPORT_VERSION = "1.3"
 
 
+def is_float(value: Any) -> bool:
+    try:
+        number = float(value)
+        return False if math.isnan(number) else True
+    except Exception:
+        return False
+
+
+def rm_df_columns(data: pd.DataFrame, pattern: str) -> pd.DataFrame:
+    new_data = data
+    for column in data.columns:
+        if re.match(pattern, column):
+            new_data = new_data.drop(columns=[column])
+    return new_data
```

Comment:
Let's only call `drop` once:

```suggestion
    columns_to_drop = []
    for column in data.columns:
        if re.match(pattern, column):
            columns_to_drop.append(column)
    return data.drop(columns=columns_to_drop)
```